packages/common/src/hooks/purchaseContent/usePurchaseContentFormConfiguration.ts

```diff
@@ -65,22 +65,33 @@ export const usePurchaseContentFormConfiguration = ({
         presetValues,
         customAmount
       })
+      const startPurchaseParams = {
+        purchaseMethod,
+        extraAmount,
+        extraAmountPreset: amountPreset,
+        contentId: track.track_id,
+        contentType: ContentType.TRACK
+      }
 
       if (purchaseMethod === PurchaseMethod.CRYPTO) {
-        openUsdcManualTransferModal()
+        openUsdcManualTransferModal({
+          isOpen: true,
+          source: 'purchase',
+          amount: price + extraAmount,
+          startPurchaseParams
```

Comment:
Can you convert this into an `onSuccessAction` instead, and have it be `onSuccessAction: startPurchaseContentFlow(startPurchaseParams)`? That makes it more general which is kinda nice

Reply:
see `openInboxUnavailableModal` for example